edb/graphql/types.py

```diff
@@ -243,6 +330,11 @@ def get_gql_name(self, name):
         module, shortname = name.split('::', 1)
         if module in {'default', 'std'}:
             if shortname.startswith('__'):
+                # Use '_edb' prefix to mark derived and otherwise
+                # internal types. We opt out of '__edb' because we
+                # still rely on the first occurrence of '__' in
```

Comment:
If it's a leading `__`, why would that be ambiguous with a module separator that would be on the interior of the name?

Reply:
Because the place where it matters the first occurrence of `__` is used as a separator for the module name. It's possible to work around it there, but it seems to be trading one awkward name for another.

Reply:
Actually, there's a simpler reason - leading `__` is reserved by GraphQL and the validator throws an exception when it sees that.